File: binance4j-convert/src/main/java/com/binance4j/convert/client/ConvertClient.java
package com.binance4j.convert.client;

import com.binance4j.convert.dto.ConversionHistory;
import com.binance4j.convert.param.ConversionParams;
import com.binance4j.core.Request;
import com.binance4j.core.client.RestClient;
import com.binance4j.core.param.Params;
import com.binance4j.core.param.TimeFrame;

/**
 * Api client for the convert endpoints
 * 
 * @see <a href= "https://binance-docs.github.io/apidocs/spot/en/#convert-endpoints">Documentation</a>
 */
public class ConvertClient extends RestClient<ConvertMapping> {
	/**
	 * @param key    API public key.
	 * @param secret API secret key.
	 */
	public ConvertClient(String key, String secret) {
		super(ConvertMapping.class, key, secret);
	}

	/**
	 * Get assets conversion history.
	 * 
	 * @param params Request params.
	 * @return The request to execute.
	 */
	public Request<ConversionHistory> getConversions(TimeFrame params) {
		return new Request<>(service.getConversions(Params.merge(new ConversionParams(), params)));
	}
}

File: src/main/kotlin/com/binance4j/connectors/convert/client/ConvertClient.kt
/*
 * MIT License
 *
 * Copyright (c) 2022 Binance4j
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.binance4j.connectors.convert.client

import com.binance4j.connectors.core.Request
import com.binance4j.connectors.core.client.RestClient

/**
 * Api client for the convert endpoints.
 *
 * [Documentation](https://binance-docs.github.io/apidocs/spot/en/#convert-endpoints)
 */
object ConvertClient : RestClient<ConvertMapping>(ConvertMapping::class.java) {
	/**
	 * Get assets conversion history.
	 *
	 * @param startTime Start time in ms.
	 * @param endTime   End time in ms.
	 * @param limit     Results limit.
	 *
	 * @return The request to execute.
	 */
	@JvmOverloads
	fun getConversions(startTime: Long, endTime: Long, limit: Int? = null) = Request(service.getConversions(startTime, endTime, limit))
}